utils/client.go

```diff
@@ -252,37 +252,26 @@ func (c *Client) RemoveServer(datacenterId, serverId string) error {
 }
 
 func (c *Client) CreateAttachVolume(datacenterId, serverId string, volProperties *ClientVolumeProperties) (*sdkgo.Volume, error) {
-	var inputVolume sdkgo.Volume
-	// TODO: if in if !!! Return early instead...
-	if volProperties != nil {
-		if volProperties.ImageId != "" {
-			inputVolume = sdkgo.Volume{
-				Properties: &sdkgo.VolumeProperties{
-					Type:             &volProperties.DiskType,
-					Size:             &volProperties.DiskSize,
-					Name:             &volProperties.Name,
-					Image:            &volProperties.ImageId,
-					ImagePassword:    &volProperties.ImagePassword,
-					SshKeys:          &[]string{volProperties.SshKey},
-					UserData:         &volProperties.UserData,
-					AvailabilityZone: &volProperties.Zone,
-				},
-			}
-		} else {
-			inputVolume = sdkgo.Volume{
-				Properties: &sdkgo.VolumeProperties{
-					Type:             &volProperties.DiskType,
-					Size:             &volProperties.DiskSize,
-					Name:             &volProperties.Name,
-					ImageAlias:       &volProperties.ImageAlias,
-					ImagePassword:    &volProperties.ImagePassword,
-					SshKeys:          &[]string{volProperties.SshKey},
-					UserData:         &volProperties.UserData,
-					AvailabilityZone: &volProperties.Zone,
-				},
-			}
-		}
+	if volProperties == nil {
+		return nil, fmt.Errorf("volume properties are nil")
+	}
+	var inputProperties sdkgo.VolumeProperties
+	inputProperties.Type = &volProperties.DiskType
+	inputProperties.Size = &volProperties.DiskSize
+	inputProperties.ImagePassword = &volProperties.ImagePassword
+	inputProperties.SshKeys = &[]string{volProperties.SshKey}
+	inputProperties.AvailabilityZone = &volProperties.Zone
+
+	if volProperties.ImageId != "" {
```

Comment:
what is none is provided?

Reply:
- if the user doesn't touch this flag at all, ubuntu:latest is set as ImageAlias
- if the user explicitly sets an empty string, an error would be thrown in PreCreateChecks